### ISSUE TITLE
accessibility: impossible to add or edit a post's title with Jaws for Windows screen reader activated

### ISSUE THREAD

**Issue by elettrona** [WordPress/gutenberg#35440]
### Description

When adding a post, page, or other custom contents, it's impossible to edit the post's title in a machine running JAWS for Windows screen reader - www.freedomscientific.com -
Expected behaviour: when adding a post, the focus is brought to title and screen reader activated editing mode, in order to write the post title. And, pressing "e" - the screen reader's shortcut to access the first edit box -, should take to the Title field.
Current behaviour: when screen reader JAWS is activated, the "add title" is read as a heading level 1, - h1 html tag -, but it can not be edited. This occurs when adding or editing a post or page, in full editing mode; only way to place a title in a content, is saving it as a draft and then adding title by the "quick editing" from "all posts" section. Visually, the focus is correctly brought to the title and it can also be written, but screen reader gives no feedback.

### Step-by-step reproduction instructions

0. Requirements: JAWS for Windows screen reader demo is required; other screen readers such as the free NVDA and VoiceOver for Apple systems, seem not affected by the issue.
1. with JAWS turned on, go to WordPress and add a new post.
2. Press Tab in order to focus the title field.
3. It will go to the block editor directly.
4. Deactivate the virtual PC cursor with insert+z key.
5. From the beginning of the window, tab till you should visually focus the title field - the screen reader will say Title, but even if you can see what you write, nothing appears to voice feedback.
6. alternative: when opening the Add new post window, press the "e" letter on the keyboard to focus the first edit box. It should take to title directly, it goes to the content instead!

### Environment info

WordPress 5.8.1, Gutenberg 11.6.0 development version, GeneratePress theme. Windows 10 on a desktop with latest build of JAWS 2021 screen reader. 

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by alexstine** [comment]
Hello @talksina ,

Thank you for the report.

I think what is happening is you may not be waiting enough time for JAWS to focus in the title field. E.g. when you select Add Post, the default behavior is to focus the Add Title field. However, I do agree that this implementation is a problem and I find it quite clunky with the latest JAWS version. I'm going to open a pull request here shortly which should hopefully improve this for a future release.

@ellatrix I think this is the start of this issue.

https://github.com/WordPress/gutenberg/commit/647547a7ab9947d212a8ba89a6054d91b546174a#diff-0f52c17f88643236b2ac758d462a5a00e7b7b6a52b2eb49936cae64070e2aaf3

This commit essentially makes a non-native input element a text input. I'm not sure why this was changed but I notice that I also have difficulty using JAWS. It's not as bad as this user is describing, but I think older JAWS versions don't handle contentEditable very well. I also notice that NVDA doesn't work as expected either. The screen readers are getting confused because it is an editable heading. This has got to be really confusing users as well. Any chance maybe a separate discussion could be opened up about adding this component back or maybe using a native HTML field?

I will link my PR to this shortly and hopefully that will fix things up for the interim.

Thanks.

**Comment by elettrona** [comment]
Hello, I sent an issue to Gutenberg and not to WordPress as if I
deactivate the Gutenberg 11.x plug-in it seems to work. But maybe this
issue can be solved adding a proper ARIA role to the title "editable
heading"?

**Comment by alexstine** [comment]
@talksina Would you be interested in testing the fix? You can test the fix at this link.

http://gutenberg.run/35898

You can also install this .zip file of the latest plugin. I've attached it to my comment.

[gutenberg.zip](https://github.com/WordPress/gutenberg/files/7406051/gutenberg.zip)

Thanks.

**Comment by elettrona** [comment]
Hello! Both tested on your side, and downloading the ZIP file on my
staging platform.
It seems to work now!
I tested on posts and pages - in the stage i have not installed other
plugins such as glossary or other fun.
Let me know about further developments!

**Comment by alexstine** [comment]
@talksina Thanks for testing. The PR has been merged and I am hoping it will make it in to the upcoming WordPress release. Don't quote me on it though as I am still unsure how the release schedules work with Gutenberg.

**Comment by elettrona** [comment]
WordPress auto-update has brought 11.8.0 Gutenberg on my web site. But
the title bug still persists.


2021-10-26 16:20 GMT+02:00, Alex Stine ***@***.***>:

**Comment by alexstine** [comment]
@talksina I believe it will be in version 11.9.

Thanks.
